Lesson5/main.js

```diff
@@ -0,0 +1,24 @@
+const ul = document.getElementById("js-ul");
+const fragment = new DocumentFragment();
+const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}];
+
+const checkFeatures = new Promise((resolve) => {
+  resolve(features);
+}).then(
+  function(value){
+    for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    }
```

Comment:
MUST
The array 'features' used in this looping process is not the value passed in Promise.
It is the global variable 'features'.

The value 'value' received in the Promise must be used in the looping process.

Reply:
Mai-san,
Thank you for your review.
I replaced the features in the for..of loop with the value passed from resolve process.
Kindly check my latest commit c576e4820a8da43a7de33e2a967e70fafc0ca58e